### PR TITLE
use xargs to check for formatting

### PR DESCRIPTION
**Pull request by mwhooker**
Pass go files to gofmt with xargs to avoid command line limits

should address issues in #6227, #4494

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -5,6 +5,7 @@ GITSHA:=$(shell git rev-parse HEAD)
 # Get the current local branch name from git (if we can, this may be blank)
 GITBRANCH:=$(shell git symbolic-ref --short HEAD 2>/dev/null)
 GOFMT_FILES?=$$(find . -not -path "./vendor/*" -name "*.go")
+BAD_FILES=$(shell echo $(GOFMT_FILES) | xargs gofmt -s -l)
```
Comment (tb3088) [inline Makefile]: The 'echo' alone can still run off the end of the Command line length restriction.

File: Makefile [hunk 2]
```diff
@@ -5,6 +5,7 @@ GITSHA:=$(shell git rev-parse HEAD)
 # Get the current local branch name from git (if we can, this may be blank)
 GITBRANCH:=$(shell git symbolic-ref --short HEAD 2>/dev/null)
 GOFMT_FILES?=$$(find . -not -path "./vendor/*" -name "*.go")
```
Comment (tb3088) [inline Makefile]: find . -path ./vendor -prune -o -name "*.go"
is "more correct".

Comment (tb3088) [reply Makefile]: doh! 
That's very odd behavior from gofmt isn't it? I can't think of a text processing command (grep, sed, awk, etc)  that when $1 is a directory takes it upon itself to process all files inside let alone recurse! Hmm, gofmt can't even handle combined flags?!

File: Makefile (hunk 1 above)
Comment (mwhooker) [inline Makefile]: should come up with a better name. this was meant to be temporary

File: Makefile (hunk 2 above)
Comment (mwhooker) [reply Makefile]: yeah gofmt is a bit of an odd duck... Most go tools now let you do something like `./...` to mean every go file not in the vendor directory, but they don't want to support that with gofmt on the bet that vendor directories will disappear. https://github.com/golang/go/issues/22173

> Without an explicit path, it processes the standard input. Given a file, it operates on that file; given a directory, it operates on all .go files in that directory, recursively. (Files starting with a period are ignored.) By default, gofmt prints the reformatted sources to standard output.

https://golang.org/cmd/gofmt/

**Comment by mwhooker** [comment]
let me know if this works for you @tb3088

**Comment by tb3088** [comment]
please see my branch instead. tb3088/packer/gocheckxargs since it controls line lengths.

**Pull request merged**